lib/rules/no-underscore-dangle.js

```diff
@@ -162,14 +166,57 @@ module.exports = {
             }
         }
 
+        /**
+         * Check if method declaration has a underscore at the end
+         * @param {ASTNode} node node to evaluate
+         * @returns {void}
+         * @private
+         */
+        function checkForTrailingUnderscoreInMethodDefinition(node) {
+            const identifier = node.key.name;
+
+            if (typeof identifier !== "undefined" && hasTrailingUnderscore(identifier) && enforceInMethodNames) {
+                context.report({
+                    node,
+                    message: "Unexpected dangling '_' in '{{identifier}}'",
+                    data: {
+                        identifier
+                    }
+                });
+            }
+        }
+
+        /**
+         * Check if method property has a underscore at the end
+         * @param {ASTNode} node node to evaluate
+         * @returns {void}
+         * @private
+         */
+        function checkForTrailingUnderscoreInMethodProperty(node) {
+            const identifier = node.key.name;
+            const isMethod = node.method;
+
+            if (typeof identifier !== "undefined" && hasTrailingUnderscore(identifier) && enforceInMethodNames && isMethod) {
```

Comment:
ah, wait, I think I remembered why this is different. We don't want to report on any property, but only on method properties. So for instance:

```javascript
const obj = {
  _foo: 'bar', // this is fine
  _bar() { }   // this will trigger the rule
}
```

The only discriminant between the two is the `method` property on the `Property` node.